solutions/ex3/Main.hs

```diff
@@ -202,6 +216,22 @@ createSortInclusionInfer ''VarL    ''LhsL
 createSortInclusionInfer ''ExpL    ''RhsL
 createSortInclusionInfer ''AssignL ''StatementL
 
+--- Question: why is it enough to just convert AssignL to StatementL as sorts?
+-- An `Assign` constructor has more data than a `Statement` constructor.
+-- A `Block` constructor has more data than a `BlockStmt` constructor.
```

Comment:
False for the same reason.

It has more holes than a BlockStmt constructor, but one of those holes is configured to only be fillable with a unit type (namely EmptyBlockEnd, which is isomorphic to the 1 type).